### PR TITLE
Granting relayer deploy

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**

### CODE REVIEW

File: universal-login-example/.eslintrc.json [hunk 1]
```diff
@@ -1,7 +1,9 @@
 {
     "env": {
         "browser": true,
-        "es6": true
+        "es6": true,
+        "node": true,
```
Comment (marekkirejczyk) [inline universal-login-example/.eslintrc.json]: Node does not sound right in react/front-end application

File: universal-login-example/package.json [hunk 2]
```diff
@@ -24,13 +24,15 @@
...
-    "build": "yarn contracts:build && parcel build ./src/index.html",
+    "build": "yarn contracts:build && yarn relayer:build && parcel build ./src/index.html",
     "lint": "eslint src/**/*.js src/**/*.jsx scripts/**/*.js",
     "test": "mocha --recursive --require babel-register test -t 10000",
     "clean": "rm -fr ./node_modules &&rm -fr ./cache && rm -fr ./build && rm -fr ./dist",
     "contracts:build": "waffle .waffle.json",
     "contracts:deploy": "babel-node scripts/deployContracts.js",
-    "dev:start": "babel-node scripts/dev-start.js"
+    "dev:start": "babel-node scripts/dev-start.js",
+    "relayer:build": "babel src/relayer --out-dir dist/relayer",
```
Comment (marekkirejczyk) [inline universal-login-example/package.json]: One build process seems to be sufficent

File: universal-login-example/src/relayer/TokenGrantingRelayer.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import Relayer from 'universal-login-relayer';
+import ethers, {Wallet, utils} from 'ethers';
+
+class TokenGrantingRelayer extends Relayer {
+  constructor(config, provider = '', deployerPrivateKey = '', tokenContractAddress = '') {
+    super(config, provider);
+    this.provider = provider || new ethers.providers.JsonRpcProvider(config.jsonRpcUrl, config.chainSpec);
+    this.deployerPrivateKey = deployerPrivateKey || config.privateKey;
+    this.tokenContractAddress = tokenContractAddress || config.tokenContractAddress;
+    this.deployerWallet = new Wallet(this.deployerPrivateKey, this.provider);
```
Comment (marekkirejczyk) [inline universal-login-example/src/relayer/TokenGrantingRelayer.js]: In super, this.provider is already initialized in exactly the same way: 
`this.provider = provider || new ethers.providers.JsonRpcProvider(config.jsonRpcUrl, config.chainSpec);`


    this.wallet = new ethers.Wallet(config.privateKey, this.provider);

File: universal-login-example/src/relayer/TokenGrantingRelayer.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import {waitForContractDeploy} from './utils';
+import Identity from 'universal-login-contracts/build/Identity';
+import Token from '../../build/Token';
+import Relayer from 'universal-login-relayer';
+import ethers, {Wallet, utils} from 'ethers';
+
+class TokenGrantingRelayer extends Relayer {
+  constructor(config, provider = '', deployerPrivateKey = '', tokenContractAddress = '') {
```
Comment (marekkirejczyk) [inline universal-login-example/src/relayer/TokenGrantingRelayer.js]: 2) You can skip default values, then arguments will be undefined and lazy loading (|| operator) will work as expected.

File: universal-login-example/src/relayer/utils.js [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
 
-const waitForContractDeploy = async (providerOrWallet, contractJSON, transactionHash, tick = 1000) => {
+const sleep = (ms) =>
+  new Promise((resolve) => setTimeout(resolve, ms));
+
+const waitForContractDeploy = async (providerOrWallet, contractJSON, tansactionHash, tick = 1000) => {
   const provider = providerOrWallet.provider ? providerOrWallet.provider : providerOrWallet;
   const abi = contractJSON.interface;
-  let receipt = await provider.getTransactionReceipt(transactionHash);
+  let receipt = await provider.getTransactionReceipt(tansactionHash);
```
Comment (marekkirejczyk) [inline universal-login-example/src/relayer/utils.js]: typo

File: universal-login-example/src/relayer/utils.js [hunk 6]
```diff
@@ -1,12 +1,15 @@
...
   while (!receipt) {
     sleep(tick);
-    receipt = await provider.getTransactionReceipt(transactionHash);
+    receipt = await provider.getTransactionReceipt(tansactionHash);
```
Comment (marekkirejczyk) [inline universal-login-example/src/relayer/utils.js]: typo

File: universal-login-example/test/relayer/TokenGrantingRelayer.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        chainId: 0},
+      ensRegistrars: {
+        'mylogin.eth': {
+          registrarAddress: ensBuilder.registrars[defaultDomain].address,
+          resolverAddress: ensBuilder.resolver.address,
+          privateKey: deployerPrivateKey
+        }
+      }
+    });
+    relayer = new TokenGrantingRelayer(config, provider, deployerPrivateKey, tokenContract.address);
```
Comment (marekkirejczyk) [inline universal-login-example/test/relayer/TokenGrantingRelayer.js]: Move `deployerPrivateKey` and `tokenContract` to `config`.

File: universal-login-example/test/relayer/TokenGrantingRelayer.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    relayer.start();
+    relayer.addHooks();
+    [identityPrivateKey, identityContractAddress] = await sdk.create('ja.mylogin.eth');
+    expectedIdentityTokenBalance = (await tokenContract.balanceOf(identityContractAddress)).add(utils.parseEther('100'));
+  });
+
+ 
+  describe('Token granting', async () => {
+    it('Should transfer tokens, when identity created', async () => {
+      await sleep(500);
```
Comment (marekkirejczyk) [inline universal-login-example/test/relayer/TokenGrantingRelayer.js]: uuugly!

better do: 
`waitUnitl(() => await tokenContract.balanceOf(identityContractAddress) === expectedIdentityTokenBalance)`

Where `waitUnitl` works in similar way to `waitForContractDeploy`, but more generalized.

Comment (JustynaBroniszewska) [reply universal-login-example/test/relayer/TokenGrantingRelayer.js]: ✅

File: universal-login-example/test/relayer/TokenGrantingRelayer.js (hunk 7 above)
Comment (JustynaBroniszewska) [reply universal-login-example/test/relayer/TokenGrantingRelayer.js]: ✅

File: universal-login-example/src/relayer/utils.js (hunk 6 above)
Comment (JustynaBroniszewska) [reply universal-login-example/src/relayer/utils.js]: ✅

File: universal-login-example/src/relayer/utils.js (hunk 5 above)
Comment (JustynaBroniszewska) [reply universal-login-example/src/relayer/utils.js]: ✅

File: universal-login-example/src/relayer/TokenGrantingRelayer.js (hunk 4 above)
Comment (JustynaBroniszewska) [reply universal-login-example/src/relayer/TokenGrantingRelayer.js]: ✅

File: universal-login-example/src/relayer/TokenGrantingRelayer.js (hunk 3 above)
Comment (JustynaBroniszewska) [reply universal-login-example/src/relayer/TokenGrantingRelayer.js]: ✅

**Pull request merged**